pkg/test/apps_http_handler_test.go

```diff
@@ -0,0 +1,42 @@
+// +build integration
```

Comment:
The dir structure here shows not to be properly organized. Here you add it in a test dir, however, you also add tests on  `pkg/web/apps/apps_psql_repository_test.go`. 

IHMO we should organize it by layers and not by business model. For example:
/handler
/service
/repository

Following the main reasons:

1. It makes clear for the devs the architecture defined to the project
2. Ita void confusion and help the dev not to do implementation which thoughts the layers
3. It makes easier the understand and where we should looking for or add the files

Also, shows that in go the test usually stays with the implementations. So, for example, the file app_handler_test will be together with app_handler and both in the dir /handler if my suggestion is accepted by the team.

Reply:
@camilamacedo86 this is outside of the scope of this PR. We have already agreed that the current approach is the best one.

[Trying Clean Architecture on Golang](https://hackernoon.com/golang-clean-archithecture-efd6d7c43047)

Reply:
@craicoverflow the project shows not following what has in the Blog Post added for you too. See it here https://hackernoon.com/trying-clean-architecture-on-golang-2-44d615bf8fdf. 

<img width="512" alt="screenshot 2019-02-18 at 16 10 36" src="https://user-images.githubusercontent.com/7708031/53007352-fa0ff300-342e-11e9-900e-4a325628e10c.png">

IHMO: We should create a task to skill up it and review, recheck if it is all place, if not them do the changes and add the argumentation in the PR for we are able to discuss it. Wdyt @davidffrench?

Reply:
There is no need for the sub folders until we have more repositorys and delivery methods in my opinion. It just makes the folder structure more verbose in my opinion. It is unlikely we will have any additional storage or delivery methods in the future either.

Reply:
@camilamacedo86 this is not related to this PR. This can be discussed in our other communication channels if needed. We are not following the format from the blog post strictly, as you can see we are also following the [go-standard-layout](https://github.com/golang-standards/project-layout).

The main difference between ours and the blog post is they have subfolders per package. This is a valid discussion but it won't get the appropriate attention here.

Reply:
@davidffrench the main diff over what has in the blog post and it this project is not the subfolders. According to the blog shows that it should be something as  

/apps
|----repository
/versions
|----repository
/devices
|----repository
/middleware
---> here will be the routers 

instead of the pkg.

Reply:
@camilamacedo86 I get you. As Enda said above, we are using a combination of both clean architecture and go-standard-layout. The clean architecture from the blog post was one person's interpretation of a language agnostic clean architecture solution.

Reply:
@davidffrench +1. 

We are also following the **aerogear format** - see [app-metrics](https://github.com/aerogear/aerogear-app-metrics). This also follows [go-standard-layout](https://github.com/golang-standards/project-layout).